src/Sarif/ExtensionMethods.cs

```diff
@@ -173,7 +175,8 @@ public static string GetFileName(this Uri uri)
         {
             if (!uri.IsAbsoluteUri)
             {
-                throw new InvalidOperationException();
+                var newAbsoluteUri = new Uri(BaseUri, uri.OriginalString);
+                return Path.GetFileName(newAbsoluteUri.LocalPath);
```

Comment:
>GetFileName [](start = 28, length = 11)

make sure this works with a path like \blah.txt and blah.txt (no prefixed slash).